### PR TITLE
'Access-Control-Allow-Origin' response header check

### PR DESCRIPTION
**Pull request by alabiaga**
related: #12433

### CODE REVIEW

File: src/utils/xhr-utils.js [hunk 1]
```diff
@@ -376,6 +383,11 @@ export function verifyAmpCORSHeaders(win, response, init) {
     userAssert(false, 'Response must contain the' +
         ` ${ALLOW_SOURCE_ORIGIN_HEADER} header`);
   }
+  devAssert(accessControlAllowOrigin,
```
Comment (jridgewell) [inline src/utils/xhr-utils.js]: `devAssert`s will be striped from production. Also, if there were no accessControllAllowOrigin header on the response, we wouldn't have a response to interact with! Unless it opaque, maybe?

Comment (alabiaga) [reply src/utils/xhr-utils.js]: > `devAssert`s will be striped from production

Yes, I think the goal is to catch this only in early development and the message/error is to signal to the developer that CORS testing needs to be done. Which seems to be something they tend to omit.

#21313 is also another effort to make the developer more aware about CORS testing.

Comment (jridgewell) [reply src/utils/xhr-utils.js]: > to signal to the developer that CORS testing needs to be done

You're confusing `devAssert` (assert to the AMP Core developers) with `userAssert` (assert to the publisher's developers).

Comment (alabiaga) [reply src/utils/xhr-utils.js]: ah gotcha..oops. Thanks

**Comment by alabiaga** [comment]
Thanks for review @jridgewell. I still need to verify the response when that header isn't set. Thanks.

**Comment by alabiaga** [comment]
Closing @jridgewill was correct. JS doesn't have access to response data when it returns without the proper CORS headers

**Pull request closed without merging**